Neos.Fusion/Classes/FusionObjects/Helpers/LazyProps.php

```diff
@@ -43,8 +44,16 @@ final class LazyProps implements \ArrayAccess, \Iterator, \JsonSerializable
      */
     private $effectiveContext;
 
+    /**
+     * LazyProps constructor.
+     * @param ComponentImplementation $fusionObject
+     * @param string $parentPath
+     * @param Runtime $runtime
+     * @param array $keys
+     * @param array $effectiveContext
+     */
     public function __construct(
-        object $fusionObject,
+        ComponentImplementation $fusionObject,
```

Comment:
I'd rather declare `AbstractFusionObject` here (and in the property and doc comment ofc.) because the code is not actually tied to the `ComponentImplementation`.

In `Runtime` we declare `public function evaluate(string $fusionPath, $contextObject = null, string $behaviorIfPathNotFound = self::BEHAVIOR_RETURNNULL)` (with `object $contextObject` in the doc comment). So we could as well just remove `object` here.